qualtran/bloqs/rotations/hamming_weight_phasing.py

```diff
@@ -66,6 +69,7 @@ class HammingWeightPhasing(GateWithRegisters):
 
     Registers:
         x: A `THRU` register of `bitsize` qubits.
+        extra_registers: Any additional registers used by the `QvrInterface` phase oracle.
```

Comment:
document the new `phase_oracle` arg and reference the new factory method in the class docstring somewhere